src/Business Foundation/App/AuditCodes/Permissions/AuditCodesView.permissionset.al

```diff
@@ -0,0 +1,18 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+
+namespace Microsoft.Foundation.AuditCodes;
+
+permissionset 232 "Audit Codes - View"
+{
+    Assignable = false;
+    Access = Internal;
+    IncludedPermissionSets = "Audit Codes - Read";
+
+    Permissions = tabledata "Reason Code" = I,
+                tabledata "Return Reason" = I,
+                tabledata "Source Code" = I,
+                tabledata "Source Code Setup" = I;
```

Comment:
Should we grant anyone Insert permission to "Source Code Setup"?

Maybe: Inherent Insert permission on a InitRecord procedure?

Reply:
The insert would either be an empty record or the setup from InitCompany. Both are fine and yes, anyone should be allowed to do those. What is restricted is modifying an existing setup.